### PR TITLE
[PF-2504] Create cloud specific subpages under bio.terra.workspace.service.workspace.flight

### PR DESCRIPTION
**Pull request by dexamundsen**
Create cloud specific subpages under bio.terra.workspace.service.workspace.flight

Current set up in package bio.terra.workspace.service.workspace.flight
- Subpackage create.azure contains only azure specific & creation related classes
- All other gcp / azure related classes are under the parent package above

Change
- Remove subpackage create.azure 
- Create cloud specific subpackages under the parent - gcp and azure and move all cloud specific classes to respective subpackages

Reason for change
- Ease of maintenance, keeping it consistent across similar packages
---
Other changes -
- Minor formating changes in openApi.yaml
- remove the redefinition of samService in WorkspaceApiController and use the in parent class
- [PF-2510](https://broadworkbench.atlassian.net/browse/PF-2510) - replace wildcard imports by individual imports

[PF-2510]: https://broadworkbench.atlassian.net/browse/PF-2510?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: service/src/main/java/bio/terra/workspace/service/resource/ResourceValidationUtils.java [hunk 1]
```diff
@@ -203,7 +202,6 @@ public static void validateControlledResourceRegionAgainstPolicy(
       case AZURE -> {
         // TODO: enable policy check in Azure when we support Azure regions in the TPS ontology.
         // validateAzureRegion(location);
-        return;
```
Comment (yuhuyoyo) [inline service/src/main/java/bio/terra/workspace/service/resource/ResourceValidationUtils.java:206]: don't it fall through if you remove return?

File: service/src/test/java/bio/terra/workspace/service/workspace/flight/create/azure/CreateAzureContextFlightTest.java [hunk 2]
```diff
@@ -13,6 +13,7 @@
 import bio.terra.workspace.service.job.JobService;
```
Comment (yuhuyoyo) [inline service/src/test/java/bio/terra/workspace/service/workspace/flight/create/azure/CreateAzureContextFlightTest.java:13]: can you also move this test to flight/azure

File: service/src/test/java/bio/terra/workspace/service/workspace/flight/DeleteControlledAzureResourcesStepTest.java [hunk 3]
```diff
@@ -23,6 +23,7 @@
 import bio.terra.workspace.service.resource.controlled.model.ControlledResourceFields;
 import bio.terra.workspace.service.resource.controlled.model.ManagedByType;
 import bio.terra.workspace.service.resource.model.CloningInstructions;
+import bio.terra.workspace.service.workspace.flight.azure.DeleteControlledAzureResourcesStep;
```
Comment (yuhuyoyo) [inline service/src/test/java/bio/terra/workspace/service/workspace/flight/DeleteControlledAzureResourcesStepTest.java:26]: move the test to flight/azure

File: service/src/test/java/bio/terra/workspace/service/workspace/flight/DeleteGcpContextFlightTest.java [hunk 4]
```diff
@@ -22,6 +22,10 @@
 import bio.terra.workspace.service.workspace.GcpCloudContextService;
 import bio.terra.workspace.service.workspace.WorkspaceService;
 import bio.terra.workspace.service.workspace.exceptions.CloudContextRequiredException;
+import bio.terra.workspace.service.workspace.flight.gcp.CreateGcpContextFlightV2;
```
Comment (yuhuyoyo) [inline service/src/test/java/bio/terra/workspace/service/workspace/flight/DeleteGcpContextFlightTest.java:25]: move the test to flight/gcp

File: service/src/test/java/bio/terra/workspace/service/workspace/flight/DeleteAzureContextFlightTest.java [hunk 5]
```diff
@@ -25,7 +25,10 @@
 import bio.terra.workspace.service.spendprofile.SpendProfileId;
 import bio.terra.workspace.service.workspace.AzureCloudContextService;
 import bio.terra.workspace.service.workspace.WorkspaceService;
-import bio.terra.workspace.service.workspace.flight.create.azure.CreateAzureContextFlight;
```
Comment (yuhuyoyo) [inline service/src/test/java/bio/terra/workspace/service/workspace/flight/DeleteAzureContextFlightTest.java:28]: move to flight/azure

File: service/src/test/java/bio/terra/workspace/service/workspace/flight/CreateGcpContextFlightV2Test.java [hunk 6]
```diff
@@ -32,6 +32,7 @@
 import bio.terra.workspace.service.workspace.GcpCloudContextService;
 import bio.terra.workspace.service.workspace.WorkspaceService;
 import bio.terra.workspace.service.workspace.exceptions.MissingSpendProfileException;
+import bio.terra.workspace.service.workspace.flight.gcp.*;
```
Comment (yuhuyoyo) [inline service/src/test/java/bio/terra/workspace/service/workspace/flight/CreateGcpContextFlightV2Test.java:35]: please specify the import.

also move flight to flight/gcp

Comment (mikenev) [reply service/src/test/java/bio/terra/workspace/service/workspace/flight/CreateGcpContextFlightV2Test.java:35]: curious about the * import - is it against our coding standard? If so, we should add a lint rule. I know IntelliJ has collapsed imports for me before.

Comment (yuhuyoyo) [reply service/src/test/java/bio/terra/workspace/service/workspace/flight/CreateGcpContextFlightV2Test.java:35]: from java style guide: https://engdoc.corp.google.com/eng/doc/devguide/java/style/index.md?cl=head#s3-source-file-structure

File: service/src/test/java/bio/terra/workspace/service/workspace/flight/DeleteGcpContextFlightTest.java (hunk 4 above)
Comment (dexamundsen) [reply service/src/test/java/bio/terra/workspace/service/workspace/flight/DeleteGcpContextFlightTest.java]: done

File: service/src/test/java/bio/terra/workspace/service/workspace/flight/DeleteControlledAzureResourcesStepTest.java (hunk 3 above)
Comment (dexamundsen) [reply service/src/test/java/bio/terra/workspace/service/workspace/flight/DeleteControlledAzureResourcesStepTest.java]: done

File: service/src/test/java/bio/terra/workspace/service/workspace/flight/DeleteAzureContextFlightTest.java (hunk 5 above)
Comment (dexamundsen) [reply service/src/test/java/bio/terra/workspace/service/workspace/flight/DeleteAzureContextFlightTest.java:28]: done

File: service/src/test/java/bio/terra/workspace/service/workspace/flight/CreateGcpContextFlightV2Test.java (hunk 6 above)
Comment (dexamundsen) [reply service/src/test/java/bio/terra/workspace/service/workspace/flight/CreateGcpContextFlightV2Test.java]: Yes, it was automatically imported by IntelliJ. Changed and created https://broadworkbench.atlassian.net/browse/PF-2510

File: service/src/main/java/bio/terra/workspace/app/controller/WorkspaceApiController.java [hunk 7]
```diff
@@ -475,8 +473,12 @@ public ResponseEntity<Void> grantRole(
     // No additional authz check as this is just a wrapper around a Sam endpoint.
     SamRethrow.onInterrupted(
         () ->
-            samService.grantWorkspaceRole(
-                uuid, getAuthenticatedInfo(), WsmIamRole.fromApiModel(role), body.getMemberEmail()),
+            getSamService()
```
Comment (mikenev) [inline service/src/main/java/bio/terra/workspace/app/controller/WorkspaceApiController.java:476]: Why do we need to getSamService() if it is AutoWired?

Comment (dexamundsen) [reply service/src/main/java/bio/terra/workspace/app/controller/WorkspaceApiController.java:476]: Autowire annotation is used for Dependency Injection -> it injects required beans (samService) to the current bean (WorkspaceApiController) for its construction.

In other words, this annotation provides the objects as constructor args at line 113 (note: this constructor is not manually called anywhere and changing the arg order does not cause an error).

After construction the external beans are not available and a ref needs to be stored during construction if needed. The parent class has this field as a private member, and hence need the getter to access it in the child class

**Review by mikenev — APPROVED** [review]

**Review by yuhuyoyo — APPROVED** [review]

**Pull request merged**